indexing-service/src/main/java/org/apache/druid/indexing/input/DruidSegmentReader.java

```diff
@@ -105,6 +105,29 @@
   @Override
   protected CloseableIterator<Map<String, Object>> intermediateRowIterator() throws IOException
   {
+    if (source.isFromTombstone()) {
```

Comment:
Maybe don't do this here?  `DruidSegmentInputFormat` can check if the `InputEntity` is a tombstone and return an `DruidTombstoneSegmentReader` which only knows how to return empty stuff

Reply:
I read this too late before I pushed the latest change when I removed the method from the interface. But I will fix this in next commit.

Reply:
Done.